owncloudApp/src/main/java/com/owncloud/android/ui/activity/PassCodeActivity.kt

```diff
@@ -0,0 +1,425 @@
+        super.onCreate(savedInstanceState)
+
+        /// protection against screen recording
+        if (!BuildConfig.DEBUG) {
+            window.addFlags(WindowManager.LayoutParams.FLAG_SECURE)
+        } // else, let it go, or taking screenshots & testing will not be possible
+        setContentView(R.layout.passcodelock)
+        val passcodeLockLayout = findViewById<LinearLayout>(R.id.passcodeLockLayout)
+        mBCancel = findViewById(R.id.cancel)
+        mPassCodeHdr = findViewById(R.id.header)
+        mPassCodeHdrExplanation = findViewById(R.id.explanation)
+        mPassCodeError = findViewById(R.id.error)
+
+        // Allow or disallow touches with other visible windows
+        passcodeLockLayout.filterTouchesWhenObscured =
+            PreferenceUtils.shouldDisallowTouchesWithOtherVisibleWindows(this)
+        mPassCodeHdrExplanation.setFilterTouchesWhenObscured(
+            PreferenceUtils.shouldDisallowTouchesWithOtherVisibleWindows(this)
+        )
+
+
+        inflatePasscodeTxtLine(passcodeLockLayout)
+        if (ACTION_CHECK == intent.action) {
+            /// this is a pass code request; the user has to input the right value
+            mPassCodeHdr.setText(R.string.pass_code_enter_pass_code)
+            mPassCodeHdrExplanation.setVisibility(View.INVISIBLE)
+            setCancelButtonEnabled(false) // no option to cancel
+        } else if (ACTION_REQUEST_WITH_RESULT == intent.action) {
+            if (savedInstanceState != null) {
+                mConfirmingPassCode = savedInstanceState.getBoolean(KEY_CONFIRMING_PASSCODE)
+                mPassCodeDigits = savedInstanceState.getStringArray(KEY_PASSCODE_DIGITS)!!
+            }
+            if (mConfirmingPassCode) {
+                //the app was in the passcodeconfirmation
+                requestPassCodeConfirmation()
+            } else {
+                /// pass code preference has just been activated in Preferences;
+                // will receive and confirm pass code value
+                mPassCodeHdr.setText(R.string.pass_code_configure_your_pass_code)
+                //mPassCodeHdr.setText(R.string.pass_code_enter_pass_code);
+                // TODO choose a header, check iOS
+                mPassCodeHdrExplanation.setVisibility(View.VISIBLE)
+                setCancelButtonEnabled(true)
+            }
+        } else if (ACTION_CHECK_WITH_RESULT == intent.action) {
+            /// pass code preference has just been disabled in Preferences;
+            // will confirm user knows pass code, then remove it
+            mPassCodeHdr.setText(R.string.pass_code_remove_your_pass_code)
+            mPassCodeHdrExplanation.setVisibility(View.INVISIBLE)
+            setCancelButtonEnabled(true)
+        } else {
+            throw IllegalArgumentException(R.string.illegal_argument_exception_message.toString() + " ")
+        }
+        setTextListeners()
+    }
+
+    private fun inflatePasscodeTxtLine(passcodeLockLayout: LinearLayout) {
```

Comment:
`passcodeLockLayout` parameter is never used, is it needed or can we get rid of it?